### PR TITLE
add missing symbols for browser

### PR DESCRIPTION
**Pull request by ArneBab**

### CODE REVIEW

File: js2-mode.el [hunk 1]
```diff
@@ -161,6 +161,19 @@
...
+            ;; https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope
+            ;; properties
+            caches indexedDB isSecureContext origin
+            ;; methods
+            atob btoa clearInterval clearTimeout
+            createImageBitmap fetch queueMicrotask setInterval
+            setTimeout
+            
+            ;; from jslint "browser"
+            Audio document event history Image location name
```
Comment (dgutov) [inline js2-mode.el:174]: What do `event` and `name` do?

File: js2-mode.el [hunk 2]
```diff
@@ -161,6 +161,19 @@
...
+            navigator Option screen XMLHttpRequest
```
Comment (dgutov) [inline js2-mode.el:175]: `XMLHttpRequest` is specified at the end of the existing list.

`navigator`, `screen`, `document` are also present elsewhere.

Could you maybe find specific sections for `Audio`, `Image` and `Option`?

**Comment by ArneBab** [comment]
Dmitry Gutov <notifications@github.com> writes:

> @dgutov requested changes on this pull request.
>
>
>
>> @@ -161,6 +161,19 @@
>              window alert confirm document java navigator prompt screen
>              self top requestAnimationFrame cancelAnimationFrame
>  
> +            ;; Window or WorkerGlobalScope with support in Chromium and Firefox:
> +            ;; https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope
> +            ;; properties
> +            caches indexedDB isSecureContext origin
> +            ;; methods
> +            atob btoa clearInterval clearTimeout
> +            createImageBitmap fetch queueMicrotask setInterval
> +            setTimeout
> +            
> +            ;; from jslint "browser"
> +            Audio document event history Image location name
>
> What do `event` and `name` do?

I’m not sure why these are in jslint browser. But I see that event is
auto-completed in the browser console. But there it’s "undefined". So
maybe better not … thank you for checking!

>> @@ -161,6 +161,19 @@
>              window alert confirm document java navigator prompt screen
>              self top requestAnimationFrame cancelAnimationFrame
>  
> +            ;; Window or WorkerGlobalScope with support in Chromium and Firefox:
> +            ;; https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope
> +            ;; properties
> +            caches indexedDB isSecureContext origin
> +            ;; methods
> +            atob btoa clearInterval clearTimeout
> +            createImageBitmap fetch queueMicrotask setInterval
> +            setTimeout
> +            
> +            ;; from jslint "browser"
> +            Audio document event history Image location name
> +            navigator Option screen XMLHttpRequest
>
> `XMLHttpRequest` is specified at the end of the existing list.
> `navigator`, `screen`, `document` are also present elsewhere.

I removed these.

> Could you maybe find specific sections for `Audio`, `Image` and `Option`?

I created one named HTML element constructors. Would DOM level 1 be
better? (that seems to be closer, but I’m not sure about that).

Best wishes,
Arne
-- 
Unpolitisch sein
heißt politisch sein
ohne es zu merken

**Comment by dgutov** [comment]
> I created one named HTML element constructors. Would DOM level 1 be
better? (that seems to be closer, but I’m not sure about that).

I'm also not sure. They are a part of some W3C standard, right?

**Comment by ArneBab** [comment]
Option is here: https://www.w3.org/TR/2012/WD-html5-20121025/the-option-element.html#dom-option
Same for Image: https://html.spec.whatwg.org/multipage/embedded-content.html#dom-image
And Audio: https://html.spec.whatwg.org/multipage/media.html#dom-audio

**Comment by dgutov** [comment]
> Image is a deprecated constructor

`Audio` seems to be one as well.

Okay, I'll just merge this now. Denoting the corresponding standards in the code might be more trouble than it's worth.

Thank you!

**Comment by ArneBab** [comment]
Thank you!

**Pull request merged**